src/sources/datadog_agent/logs.rs

```diff
@@ -143,13 +148,50 @@ pub(crate) fn decode_log_body(
                                 path!("ddsource"),
                                 ddsource.clone(),
                             );
-                            namespace.insert_source_metadata(
-                                source_name,
-                                log,
-                                path!("ddtags"),
-                                path!("ddtags"),
-                                ddtags.clone(),
-                            );
+
+                            if !simplify_log_tags {
+                                namespace.insert_source_metadata(
+                                    source_name,
+                                    log,
+                                    path!("ddtags"),
+                                    path!("ddtags"),
+                                    ddtags.clone(),
+                                );
+                            } else {
+                                let ts = String::from_utf8(ddtags.to_vec()).unwrap();
+                                let s = ts.split(',');
+                                let mut tags = Value::from(BTreeMap::new());
+
+                                for each in s {
+                                    // The expected format is a comma-delimited list of key-value pairs,
+                                    // which are themselves separated by colons. If no colon is present,
+                                    // then we normalize to a sentinel value. This allows us to
+                                    // handle the fact that Datadog tags are not required to have a value.
+                                    // e.g., one:yes,two:no,three
+                                    let mut kv = each.split(':');
+                                    let k = kv.next().unwrap();
+                                    match kv.next() {
+                                        Some(v) => {
+                                            tags.insert(k, v);
+                                        }
+                                        None => {
+                                            tags.insert(k, None::<Value>);
+                                        }
+                                    }
```

Comment:
This won't handle values that also contain a colon. Unless you know that that is prohibited, it would be better to use `each.find(':')` and split on that position.